spec/rspec/core_spec.rb

```diff
@@ -178,9 +178,13 @@
       ).to eq(:slow => true)
     end
   end
-
+  
+  # File.executable? always returns false in Windows
+  # In *nix operating systems, the permissions are checked against the executable bit, 
+  # but in windows os we don't have the same permissions 
+  # This project had the same issue https://projects.puppetlabs.com/issues/20302
   describe "::Core.path_to_executable" do
-    it 'returns the absolute location of the exe/rspec file', :failing_on_appveyor do
+    it 'returns the absolute location of the exe/rspec file', :if => !RSpec::Support::OS.windows? do
       expect(File.exist? RSpec::Core.path_to_executable).to be_truthy
       expect(File.executable? RSpec::Core.path_to_executable).to be_truthy
     end
```

Comment:
I like your idea better.
On Feb 3, 2015 5:16 PM, "Myron Marston" <notifications@github.com> wrote:

> In spec/rspec/core_spec.rb
> <https://github.com/rspec/rspec-core/pull/1860#discussion_r24050015>:
>
> >    describe "::Core.path_to_executable" do
> > -    it 'returns the absolute location of the exe/rspec file', :failing_on_appveyor do
> > +    it 'returns the absolute location of the exe/rspec file', :if => !RSpec::Support::OS.windows? do
> >        expect(File.exist? RSpec::Core.path_to_executable).to be_truthy
> >        expect(File.executable? RSpec::Core.path_to_executable).to be_truthy
> >      end
>
> Thanks for looking into this. I definitely didn't know that
> File.executable? is always false on windows!
>
> Instead of disabling the spec entirely on windows, I'm thinking we can
> tweak the spec so that it still does some checks on windows. Something like
> this:
>
> expect(File.exist? RSpec::Core.path_to_executable).to be_truthy
> expect(File.read(RSpec::Core.path_to_executable)).to include("RSpec::Core::Runner.invoke")unless RSpec::Support::OS.windows?
>   expect(File.executable? RSpec::Core.path_to_executable).to be_truthyend
>
> That way, we're still testing that a valid file is returned on all
> platforms, and that it's a file with a line of code that will run rspec. On
> non-windows platforms we can still check executable?.
>
> Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rspec/rspec-core/pull/1860/files#r24050015>.
>